src/components/KaspaAddress.tsx

```diff
@@ -2,6 +2,7 @@ import { FC, useMemo, useState, useEffect, useRef } from "react";
 import { Square2StackIcon } from "@heroicons/react/24/outline";
 import { toast } from "../utils/toast";
 import { useIsMobile } from "../utils/useIsMobile";
+import { useMessagingStore } from "../store/messaging.store";
 
 interface KaspaAddressProps {
   address: string | { toString: () => string };
```

Comment:
I feel like we should add explicit optional props:
```ts
{
  address?: string | { toString: () => string };
  nickname?: string;
}
```

It needs to have at least one to operate, if both, nickname take precedence?
What do you think?